module/modmanager/manager_test.go

```diff
@@ -125,6 +125,11 @@ func TestModManagerFunctions(t *testing.T) {
 	ok = mgr.IsModularResource(resource.NameFromSubtype(generic.Subtype, "missing"))
 	test.That(t, ok, test.ShouldBeFalse)
 
+	t.Log("test Validate")
+	deps, err := mgr.Validate(ctx, cfgCounter1)
```

Comment:
Does this actually pass? If so, what's it actually doing? I don't see any Validate() code being added to simplemodule (where this lives) so I'm not sure what's actually getting tested here.

Reply:
It does pass. Added a comment in the testing code to clarify, but this test is useful to test that when `ValidateConfig` is called on a resource that has no registered "configuration object" with an associated `Validate` method, the call to `ValidateConfig` does not fail. In other words, it tests [this logic](https://github.com/viamrobotics/rdk/pull/1975/files#diff-9a0a92d26bcd93d88bcfe6d94c11871002ddb11ea827c5f27c75c3238464d079R455-R456).

I've also added a `TestModManagerValidation` test that runs `complexmodule` and a real `ValidateConfig` below.